azure-mgmt-scheduler/src/main/java/com/microsoft/azure/management/scheduler/implementation/JobImpl.java

```diff
@@ -147,4 +250,289 @@ public JobImpl withState(JobState state) {
+            this.inner().properties().recurrence().schedule().monthlyOccurrences().add(day.monthlyOccurrence());
+        }
+
+        return this;
+    }
+
+    @Override
+    public JobImpl recurringEvery(JobScheduleDay... weekDays) {
+        if (this.inner().properties() == null) {
+            this.inner().withProperties(new JobPropertiesInner());
+            this.inner().properties()
+                .withRecurrence(new JobRecurrence());
+        }
+        if (this.inner().properties().recurrence().schedule() == null) {
+            this.inner().properties()
+                .recurrence()
+                .withSchedule(new JobRecurrenceSchedule());
+        }
+        resetJobScheduleDaysList();
+
+        this.inner().properties()
+            .recurrence()
+            .schedule().withMonthlyOccurrences(new ArrayList<JobRecurrenceScheduleMonthlyOccurrence>());
+        for (JobScheduleDay day : weekDays) {
+            this.inner().properties().recurrence().schedule()
+                .monthlyOccurrences().add(new JobRecurrenceScheduleMonthlyOccurrence().withDay(day));
+        }
+
+        return this;
+    }
+
+    @Override
+    public JobImpl atTheseMinutes(int... minutes) {
+        if (this.inner().properties() == null) {
+            this.inner().withProperties(new JobPropertiesInner());
+            this.inner().properties()
+                .withRecurrence(new JobRecurrence());
+        }
+        if (this.inner().properties().recurrence().schedule() == null) {
+            this.inner().properties()
+                .recurrence()
+                .withSchedule(new JobRecurrenceSchedule());
+        }
+        if (this.inner().properties().recurrence().schedule().minutes() == null) {
+            this.inner().properties()
+                .recurrence()
+                .schedule().withMinutes(new ArrayList<Integer>());
+        }
+        for (int minute : minutes) {
+            this.inner().properties().recurrence().schedule().minutes().add(minute);
+        }
+
+        return this;
+    }
+
+    @Override
+    public JobImpl atTheseHours(int... hours) {
+        if (this.inner().properties() == null) {
+            this.inner().withProperties(new JobPropertiesInner());
+            this.inner().properties()
+                .withRecurrence(new JobRecurrence());
+        }
+        if (this.inner().properties().recurrence().schedule() == null) {
+            this.inner().properties()
+                .recurrence()
+                .withSchedule(new JobRecurrenceSchedule());
+        }
+        if (this.inner().properties().recurrence().schedule().hours() == null) {
+            this.inner().properties()
+                .recurrence()
+                .schedule().withHours(new ArrayList<Integer>());
+        }
+        for (int hour : hours) {
+            this.inner().properties().recurrence().schedule().hours().add(hour);
+        }
+
+        return this;
+    }
+
+    @Override
+    public JobImpl endingNever() {
```

Comment:
along similar lines to before, it seems a helper `ensureRecurrence()` would help save a whole bunch of repeated code in the next few withers.